administrator/components/com_media/resources/scripts/components/browser/utils/utils.es6.js

```diff
@@ -0,0 +1,89 @@
+import * as types from '../../../store/mutation-types.es6';
+
+/**
+ * Handle the click event
+ * @param event
+ * @param ctx  the context
+ */
+export default function onItemClick(event, ctx) {
+  if (ctx.item.path && ctx.item.type === 'file') {
+    window.parent.document.dispatchEvent(
+      new CustomEvent('onMediaFileSelected', {
+        bubbles: true,
+        cancelable: false,
+        detail: {
+          type: ctx.item.type,
+          name: ctx.item.name,
+          path: ctx.item.path,
+          thumb: ctx.item.thumb,
+          fileType: ctx.item.mime_type ? ctx.item.mime_type : false,
+          extension: ctx.item.extension ? ctx.item.extension : false,
+          width: ctx.item.width ? ctx.item.width : 0,
+          height: ctx.item.height ? ctx.item.height : 0,
+        },
+      }),
+    );
+  }
+
+  if (ctx.item.type === 'dir') {
+    window.parent.document.dispatchEvent(
+      new CustomEvent('onMediaFileSelected', {
+        bubbles: true,
+        cancelable: false,
+        detail: {
+          type: this.item.type,
+          name: this.item.name,
+          path: this.item.path,
```

Comment:
```suggestion
          type: ctx.item.type,
          name: ctx.item.name,
          path: ctx.item.path,
```
I think this should fix the issue with folder selection